src/core/server/http/router/request.ts

```diff
@@ -201,6 +201,31 @@ export class KibanaRequest<
       options,
     };
   }
+
+  private getAuthRequired(request: Request): boolean | 'optional' {
+    const authOptions = request.route.settings.auth;
+    if (typeof authOptions === 'object') {
+      // 'try' is used in the legacy platform
+      if (authOptions.mode === 'optional' || authOptions.mode === 'try') {
```

Comment:
It will throw en exception in case of `try` after the migration is done. I decided to add an exhaustive check to catch the options introduced in the new hapi versions if any.